tests/run/C_argc_argv-to-java_args/src/main/scala/CArgcArgvToJavaArgs.scala

```diff
@@ -0,0 +1,10 @@
+object CArgcArgvToJavaArgs {
+  def main(args: Array[String]) {

```

Comment:
Procedural or imperative style is always hard to beat if you have concerns about code size, memory consumption and eficiency.

For example, code like `for (i <- 0 until len)` create objects under the scene.

FInally, the following is a educated guess, I would have to try it to be sure.   The LLVM backends are optimized for imperative language like C/C++.  Functional style will generate IR code that will be difficult to optimize by these backend compilers.  As more functional language target LLVM, more functional specific optimization pass will be added.

Until then, I believe it's better to stick to imperative style in a heavily used library.